dotNET/VatDetails/Program.cs

```diff
@@ -0,0 +1,221 @@
+        /// <returns>Displays SOAP message on console and returns a VATResult object containing all the information</returns>
+        public static VATResult Execute(string username, string password, string afmCalledBy, string afmCalledFor)
+        {
+            HttpWebRequest request = CreateWebRequest();
+            XmlDocument soapEnvelopeXml = new XmlDocument();
+            string xml = @"<?xml version=""1.0"" encoding=""UTF-8""?>
+                <env:Envelope
+                xmlns:env=""http://schemas.xmlsoap.org/soap/envelope/""
+                xmlns:ns=""http://gr/gsis/rgwspublic/RgWsPublic.wsdl""
+                xmlns:xsi=""http://www.w3.org/2001/XMLSchema-instance""
+                xmlns:xsd=""http://www.w3.org/2001/XMLSchema""
+                xmlns:ns1=""http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-
+                1.0.xsd"">
+                <env:Header>
+                <ns1:Security>
+                <ns1:UsernameToken>
+                <ns1:Username>$username$</ns1:Username>
+                <ns1:Password>$password$</ns1:Password>
+                </ns1:UsernameToken>
+                </ns1:Security>
+                </env:Header>
+                <env:Body>
+                <ns:rgWsPublicAfmMethod>
+                <RgWsPublicInputRt_in xsi:type=""ns:RgWsPublicInputRtUser"">
+                <ns:afmCalledBy>$afmCalledBy$</ns:afmCalledBy>
+                <ns:afmCalledFor>$afmCalledFor$</ns:afmCalledFor>
+                </RgWsPublicInputRt_in>
+                <RgWsPublicBasicRt_out xsi:type=""ns:RgWsPublicBasicRtUser"">
+                <ns:afm xsi:nil=""true""/>
+                <ns:stopDate xsi:nil=""true""/>
+                <ns:postalAddressNo xsi:nil=""true""/>
+                <ns:doyDescr xsi:nil=""true""/>
+                <ns:doy xsi:nil=""true""/>
+                <ns:onomasia xsi:nil=""true""/>
+                <ns:legalStatusDescr xsi:nil=""true""/>
+                <ns:registDate xsi:nil=""true""/>
+                <ns:deactivationFlag xsi:nil=""true""/>
+                <ns:deactivationFlagDescr xsi:nil=""true""/>
+                <ns:postalAddress xsi:nil=""true""/>
+                <ns:firmFlagDescr xsi:nil=""true""/>
+                <ns:commerTitle xsi:nil=""true""/>
+                <ns:postalAreaDescription xsi:nil=""true""/>
+                <ns:INiFlagDescr xsi:nil=""true""/>
+                <ns:postalZipCode xsi:nil=""true""/>
+                </RgWsPublicBasicRt_out>
+                <arrayOfRgWsPublicFirmActRt_out xsi:type=""ns:RgWsPublicFirmActRtUserArray""/>
+                <pCallSeqId_out xsi:type=""xsd:decimal"">0</pCallSeqId_out>
+                <pErrorRec_out xsi:type=""ns:GenWsErrorRtUser"">
+                <ns:errorDescr xsi:nil=""true""/>
+                <ns:errorCode xsi:nil=""true""/>
+                </pErrorRec_out>
+                </ns:rgWsPublicAfmMethod>
+                </env:Body>
+                </env:Envelope>";
+
```

Comment:
Yes, Visual Studio has a nice utility (https://msdn.microsoft.com/en-us/library/bb628652.aspx) that can add and/or update a web service via WSDL file. Do you prefer it this way? It will create some additional files in the project.